lib/bots/versionbot.ts

```diff
@@ -435,6 +454,13 @@ export class VersionBot extends GithubBot.GithubBot {
                 throw new Error(`alreadyCommitted`);
             }
 
+            // If this was a labeling action and there's a config, check to see if there's a maintainers
+            // list and ensure the labeler was on it.
+            if ((data.action === 'labeled') && botConfig) {
+                // Note that labeling can only occur on a PRE data, hence casting.
```

Comment:
You can use [discriminated unions](https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions) to explain this to the compiler and then it'll do the casting automatically in cases like this, and give you extra safety in some others too.